projects/components/src/splitter/splitter.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@angular/core';
+import { fromEvent, Observable } from 'rxjs';
+import { map, mergeMap, takeUntil } from 'rxjs/operators';
+import { SplitterDirection } from './splitter';
+
+@Injectable()
+export class SplitterService {
+  public buildSplitterLayoutChangeObservable(
+    splitterElement: HTMLElement,
+    parentElement: HTMLElement,
+    direction: SplitterDirection
+  ): Observable<boolean> {
+    const moveOnParent$ = fromEvent<MouseEvent>(parentElement, 'mousemove');
+    const downOnHost$ = fromEvent<MouseEvent>(splitterElement, 'mousedown');
+    const upOnWindow$ = fromEvent<MouseEvent>(window, 'mouseup');
+
+    return downOnHost$.pipe(
+      mergeMap(() => moveOnParent$.pipe(takeUntil(upOnWindow$))),
+      map(event => {
+        const parentBounds = parentElement.getBoundingClientRect();
+        const prevSibling = this.getHTMLElement(splitterElement.previousElementSibling);
+        const nextSibling = this.getHTMLElement(splitterElement.nextElementSibling);
+
+        const previousElementUpdatedDimension = prevSibling
+          ? this.calculateUpdatedDimensionForPreviousElement(event, parentBounds, direction)
+          : 0;
+        const nextElementUpdatedDimension = nextSibling
+          ? this.calculateUpdatedDimensionForNextElement(previousElementUpdatedDimension, parentBounds, direction)
+          : 0;
+        const updatedDimensionData: UpdatedDimensionData[] = [];
+
+        if (prevSibling) {
+          updatedDimensionData.push({
+            element: prevSibling,
+            dimension: previousElementUpdatedDimension
+          });
+        }
+
+        if (nextSibling) {
+          updatedDimensionData.push({
+            element: nextSibling,
+            dimension: nextElementUpdatedDimension
+          });
+        }
+
+        return updatedDimensionData;
+      }),
+      map(data => this.maybeUpdateLayoutOfSiblings(data, direction))
+    );
+  }
+
+  private maybeUpdateLayoutOfSiblings(data: UpdatedDimensionData[], direction: SplitterDirection): boolean {
+    let layoutChanged = false;
```

Comment:
this doesn't really seem necessary, it's just being set if there's any data, so we could just return `data.length > 0`

Reply:
Oh yeah. I thought such a return would look weird as it didn't depend on any work that happened in the method. I can change it.

Reply:
removed